dbt/include/duckdb/macros/materializations/external.sql

```diff
@@ -45,13 +58,55 @@
     {{- create_table_as(False, temp_relation, compiled_code, language) }}
   {%- endcall %}
 
-  -- write an temp relation into file
-  {{ write_to_file(temp_relation, location, write_options) }}
-  -- create a view on top of the location
+-- write a temp relation into file
+{{ write_to_file(temp_relation, location, write_options) }}
+
+-- create a view on top of the location
   {% call statement('main', language='sql') -%}
+    {% if format == 'json' %}
+      create or replace view {{ intermediate_relation }} as (
+        select * from read_json('{{ read_location }}'
+        {%- for key, value in json_read_options.items() -%}
+          , {{ key }}=
+          {%- if value is string -%}
+            '{{ value }}'
+          {%- else -%}
+            {{ value }}
+          {%- endif -%}
+        {%- endfor -%}
+        )
+      );
+    {% elif format == 'parquet' %}
+      create or replace view {{ intermediate_relation }} as (
+        select * from read_parquet('{{ read_location }}'
+        {%- for key, value in parquet_read_options.items() -%}
+          , {{ key }}=
+          {%- if value is string -%}
+            '{{ value }}'
+          {%- else -%}
+            {{ value }}
+          {%- endif -%}
+        {%- endfor -%}
+        )
+      );
+    {% elif format == 'csv' %}
     create or replace view {{ intermediate_relation }} as (
-        select * from '{{ read_location }}'
+      select * from read_csv('{{ read_location }}'
+      {%- for key, value in csv_read_options.items() -%}
+        , {{ key }}=
+        {%- if value is string -%}
+          '{{ value }}'
+        {%- else -%}
+          {{ value }}
+        {%- endif -%}
+      {%- endfor -%}
+      )
     );
+    {% else %}
+      create or replace view {{ intermediate_relation }} as (
```

Comment:
This condition is impossible right now, right? No objection to leaving this block in, just wanted to be sure I had this right